contents/projectManagement/testDrivenDevelopment.md

```diff
@@ -0,0 +1,232 @@
+<frontmatter>
+  title: Test-Driven Development
+  header: pagetop.md
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+<div class="website-content">
+
+# Test-Driven Development
+
+Author: [Yash Chowdhary](https://github.com/yash-chowdhary)
+
+Reviewers: [Ang Ze Yu](https://github.com/ang-zeyu), [Neil Brian](https://github.com/nbriannl), [James Pang](https://github.com/jamessspanggg), [Tejas Bhuwania](https://github.com/Tejas2805)
+
+<box id="article-toc">
+
+* [Test-Driven Development](#test-driven-development)
+  * [What is Test-driven Development](#what-is-test-driven-development)
+    * [Development Cycle](#development-cycle)
+  * [Why adopt Test-Driven Development?](#why-adopt-test-driven-development)
+    * [Benefit 1: Improves Code Quality](#benefit-1-improves-code-quality)
+    * [Benefit 2: Ensures that Product Requirements are Understood](#benefit-2-ensures-that-product-requirements-are-understood)
+    * [Benefit 3: Increases Productivity](#benefit-3-increases-productivity)
+  * [Caveats and Limitations](#caveats-and-limitations)
+  * [Advanced Test-Driven Development](#advanced-test-driven-development)
+  * [Getting Started with TDD](#getting-started-with-tdd)
+    * [Basic](#basic)  
+    * [Intermediate](#intermediate)
+    * [Advanced](#advanced)
+  * [References](#references)
+
+{.px-3}
+
+</box>
+
+## What is Test-driven Development? 
+
+Typically, when you write unit tests, you test out the functionality of code you've already written. Test-driven development (a.k.a TDD) is a software development process that emphasizes the opposite - writing tests before production code. It is a design technique aimed at understanding requirements before writing functional code.
```

Comment:
>Test-driven development (a.k.a TDD) is a software development process that emphasizes the opposite - writing tests before production code.

This implies the TDD is the opposite of the previous sentence, but that's not the case here. Something like this?
>Typically, programmer write tests _after_ they have written the code. In contrast, Test-Driven Development (a.k.a TDD) advocates writing tests _before_ the code.

Also, you can quote the definition of TDD for a credible source instead of writing your own definition.

Reply:
Pick one term and stick to it. e.g.,
* functionality, production code
* development process, design technique

Reply:
>understanding requirements before writing functional code

Sounds like you are picking one of the benefits and using that to _define_ the technique.

Reply:
> This implies the TDD is the opposite of the previous sentence, but that's not the case here. Something like this?
> Also, you can quote the definition of TDD for a credible source instead of writing your own definition.

Will edit the first paragraph to make it clearer. I have used Robert Martin as a source in the "development cycle" section. I feel that adding another source for the definition of TDD will be repetitive especially because the "development cycle" section comes right after this.